src/types.rs

```diff
@@ -586,156 +456,107 @@ pub trait Interface: Send + Sync + InterfaceClone {
         raw_coins: u64,
         data: &[u8],
         filter: Option<(&str, Option<&[u8]>)>,
-    ) -> Result<()> {
-        unimplemented!("send_message")
-    }
+    ) -> Result<()>;
 
     // Returns the operation id that originated the current execution if there
     // is one
-    fn get_origin_operation_id(&self) -> Result<Option<String>> {
-        unimplemented!("get_origin_operation_id")
-    }
+    fn get_origin_operation_id(&self) -> Result<Option<String>>;
 
     // Sha256 hash bytes
-    fn hash_sha256(&self, bytes: &[u8]) -> Result<[u8; 32]> {
-        unimplemented!("hash_sha256")
-    }
+    fn hash_sha256(&self, bytes: &[u8]) -> Result<[u8; 32]>;
 
     // Keccak256 hash bytes
-    fn hash_keccak256(&self, bytes: &[u8]) -> Result<[u8; 32]> {
-        unimplemented!("hash_keccak256")
-    }
+    fn hash_keccak256(&self, bytes: &[u8]) -> Result<[u8; 32]>;
 
     fn native_amount_from_str_wasmv1(
         &self,
         amount: &str,
-    ) -> Result<NativeAmount> {
-        unimplemented!("native_amount_from_str_wasmv1");
-    }
+    ) -> Result<NativeAmount>;
 
     fn native_amount_to_string_wasmv1(
         &self,
         amount: &NativeAmount,
-    ) -> Result<String> {
-        unimplemented!("native_amount_to_string_wasmv1");
-    }
+    ) -> Result<String>;
 
-    fn check_native_amount_wasmv1(
-        &self,
-        amount: &NativeAmount,
-    ) -> Result<bool> {
-        unimplemented!("check_native_amount_wasmv1");
-    }
+    fn check_native_amount_wasmv1(&self, amount: &NativeAmount)
+        -> Result<bool>;
 
-    fn add_native_amounts_wasmv1(
+    fn add_native_amount_wasmv1(
         &self,
         amount1: &NativeAmount,
         amount2: &NativeAmount,
-    ) -> Result<NativeAmount> {
-        unimplemented!("add_native_amounts_wasmv1");
-    }
+    ) -> Result<NativeAmount>;
 
-    fn sub_native_amounts_wasmv1(
+    fn sub_native_amount_wasmv1(
         &self,
         amount1: &NativeAmount,
         amount2: &NativeAmount,
-    ) -> Result<NativeAmount> {
-        unimplemented!("sub_native_amounts_wasmv1");
-    }
+    ) -> Result<NativeAmount>;
 
     fn mul_native_amount_wasmv1(
         &self,
         amount: &NativeAmount,
         factor: u64,
-    ) -> Result<NativeAmount> {
-        unimplemented!("mul_native_amount_wasmv1");
-    }
+    ) -> Result<NativeAmount>;
```

Comment:
Isn't this one supposed to be title as scalar?

Reply:
it's more consistent with it.
I'm doing it

Reply:
done